lib/web/databases.go

```diff
@@ -488,6 +491,44 @@ func (h *Handler) dbConnect(
 		return nil, trace.Wrap(err)
 	}
 
+	databases, err := apiclient.GetAllResources[types.DatabaseServer](ctx, clt, &proto.ListResourcesRequest{
+		Namespace:           apidefaults.Namespace,
+		ResourceType:        types.KindDatabaseServer,
+		PredicateExpression: fmt.Sprintf(`name == "%s"`, req.ServiceName),
+		Limit:               1,
+	})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	var db types.Database
+	if len(databases) > 0 {
+		db = databases[0].GetDatabase()
+	}
+	if db != nil && db.GetProtocol() == req.Protocol {
+		// As a convenience, we will append the `@<project-id>.iam` suffix to a username
+		// when connecting to Postgres GCP databases and the domain is missing.
+		//
+		// Commonly, the service account and the database share the same project ID,
+		// which means we can try to guess the intended suffix.
+		//
+		// This is only applied for Postgres (CloudSQL Postgres or AlloyDB) because:
+		// - CloudSQL MySQL still supports "classical" (one-time password) users;
+		//   otherwise it would have used the `@<project>.iam.gserviceaccount.com` suffix.
+		// - Spanner applies the `@<project>.iam.gserviceaccount.com` suffix directly in the engine.
+		if db.GetProtocol() == defaults.ProtocolPostgres &&
+			req.DatabaseUser != "" && !strings.Contains(req.DatabaseUser, "@") &&
```

Comment:
nit: should we make this a helper. seems duplicated code with `tsh`. also we may have similar cases like this in the future.